model/vcs/metrics.yaml

```diff
@@ -0,0 +1,147 @@
+    instrument: updowncounter
+    unit: "{change}"
+    stability: experimental
+    attributes:
+      - ref: vcs.change.state
+        requirement_level: required
+      - ref: vcs.repository.url.full
+        requirement_level: required
+  - id: metric.vcs.change.duration
+    type: metric
+    metric_name: vcs.change.duration
+    brief: 'The time duration a change (pull request/merge request/changelist) has been in a given state.'
+    instrument: gauge
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+      - ref: vcs.change.state
+        requirement_level: required
+  - id: metric.vcs.change.time_to_approval
+    type: metric
+    metric_name: vcs.change.time_to_approval
+    brief: 'The amount of time it took a change (pull request/merge request/changelist) to go from open to approved'
+    instrument: gauge
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+  - id: metric.vcs.repository.count
+    type: metric
+    metric_name: vcs.repository.count
+    brief: 'The number of repositories in an organization'
+    instrument: updowncounter
+    unit: "{repository}"
+    stability: experimental
+    attributes: []
+  - id: metric.vcs.ref.count
+    type: metric
+    metric_name: vcs.ref.count
+    brief: 'The number of refs of type branch or tag in a repository'
+    instrument: updowncounter
+    unit: "{ref}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.type
+        requirement_level: required
+  - id: metric.vcs.change.lines_delta
+    type: metric
+    metric_name: vcs.change.lines_delta
+    brief: 'The number of lines added/removed in a change relative to the target branch of the change'
+    instrument: gauge
+    unit: "{line}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+      - ref: vcs.ref.type
+        requirement_level: required
+      - ref: vcs.line_change.type
+        requirement_level: required
+  - id: metric.vcs.ref.lines_delta
+    type: metric
+    metric_name: vcs.ref.lines_delta
+    brief: 'The number of lines added/removed in a ref (branch) relative to the previous day'
+    instrument: gauge
+    unit: "{line}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+      - ref: vcs.ref.type
+        requirement_level: required
+      - ref: vcs.line_change.type
```

Comment:
Originally, these were two different metrics of `lines_added` & `lined_deleted`. If we continue to keep  these in one metric, then I think these would be the only types allowed.

Reply:
Cool. so then we'd report two measurements (with different dimension values), right? Then we can't use negative numbers - that'd result in `24`  instead of `+36, -12`.

So we need the attribute, and the only ask is to add a note on the metric and tell that you want instrumentation to report measurements of each type every time it's measured.

Reply:
I such a added note for the line and revision metrics in 5d5a56b2776f2da0323add3f92c5740917f0204c.
I also specified that the measurements should be positive.